WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListActivity.java

```diff
@@ -36,39 +36,75 @@ public void onCreate(Bundle savedInstanceState) {
 
         setContentView(R.layout.post_list_activity);
 
-        mIsPage = getIntent().getBooleanExtra(EXTRA_VIEW_PAGES, false);
-
         Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
         setSupportActionBar(toolbar);
 
-        ActionBar actionBar = getSupportActionBar();
-        if (actionBar != null) {
-            actionBar.setTitle(getString(mIsPage ? R.string.pages : R.string.posts));
-            actionBar.setDisplayShowTitleEnabled(true);
-            actionBar.setDisplayHomeAsUpEnabled(true);
-        }
-
         if (savedInstanceState == null) {
             mSite = (SiteModel) getIntent().getSerializableExtra(WordPress.SITE);
         } else {
             mSite = (SiteModel) savedInstanceState.getSerializable(WordPress.SITE);
         }
 
+        handleIntent(getIntent());
+    }
+
+    @Override
+    protected void onNewIntent(Intent intent) {
+        super.onNewIntent(intent);
+        handleIntent(intent);
+    }
+
+    private void handleIntent(Intent intent) {
+        mIsPage = intent.getBooleanExtra(EXTRA_VIEW_PAGES, false);
+
+        // get new intent extras and compare whether the running instance of PostsListActivity has
+        // the same values or not. If not, we need to create a new fragment and show the corresponding
+        // requested content
+        boolean bPageHasChanged = false;
```

Comment:
What's the reason for the Hungarian-style type prefix here? We don't use that anywhere else to my knowledge, could we just use `pageHasChanged`?